apis/tempo/v1alpha1/tempomonolithic_types.go

```diff
@@ -0,0 +1,265 @@
+
+	// HTTP defines the OTLP/HTTP configuration
+	//
+	// +kubebuilder:validation:Optional
+	HTTP *MonolithicIngestionOTLPProtocolsHTTPSpec `json:"http,omitempty"`
+}
+
+// MonolithicIngestionOTLPProtocolsGRPCSpec defines the settings for OTLP ingestion over GRPC.
+type MonolithicIngestionOTLPProtocolsGRPCSpec struct {
+	// Enabled defines if OTLP over gRPC is enabled
+	//
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"enabled"`
+}
+
+// MonolithicIngestionOTLPProtocolsHTTPSpec defines the settings for OTLP ingestion over HTTP.
+type MonolithicIngestionOTLPProtocolsHTTPSpec struct {
+	// Enabled defines if OTLP over HTTP is enabled
+	//
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"enabled"`
+}
+
+// MonolithicIngestionTLSSpec defines the TLS settings for ingestion.
+type MonolithicIngestionTLSSpec struct {
+	// Enabled defines if TLS is enabled for ingestion
+	//
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"enabled"`
+
+	// CA defines the name of a secret containing the CA certificate
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	CA string `json:"ca"`
+
+	// Cert defines the name of a secret containing the TLS certificate and private key
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Cert string `json:"cert"`
+}
+
+// MonolithicJaegerUISpec defines the settings for the Jaeger UI.
+type MonolithicJaegerUISpec struct {
+	// Enabled defines if the Jaeger UI should be enabled
+	//
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"enabled"`
+
+	// Ingress defines the ingress configuration for Jaeger UI
+	//
+	// +kubebuilder:validation:Optional
+	Ingress *MonolithicJaegerUIIngressSpec `json:"ingress,omitempty"`
+
+	// Route defines the route configuration for Jaeger UI
+	//
+	// +kubebuilder:validation:Optional
+	Route *MonolithicJaegerUIRouteSpec `json:"route,omitempty"`
+}
+
+// MonolithicJaegerUIIngressSpec defines the settings for the Jaeger UI ingress.
+type MonolithicJaegerUIIngressSpec struct {
+	// Enabled defines if an Ingress object should be created for Jaeger UI
+	//
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"enabled"`
+}
+
+// MonolithicJaegerUIRouteSpec defines the settings for the Jaeger UI route.
+type MonolithicJaegerUIRouteSpec struct {
+	// Enabled defines if a Route object should be created for Jaeger UI
+	//
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"enabled"`
+}
+
+// MonolithicObservabilitySpec defines the observability settings of the Tempo deployment.
+type MonolithicObservabilitySpec struct {
+	// Metrics defines the metrics configuration of the Tempo deployment
+	//
+	// +kubebuilder:validation:Optional
+	Metrics *MonolithicObservabilityMetricsSpec `json:"metrics,omitempty"`
+}
+
+// MonolithicObservabilityMetricsSpec defines the metrics settings of the Tempo deployment.
+type MonolithicObservabilityMetricsSpec struct {
```

Comment:
Can't we reuse smth from the microservice type? 

e.g. the whole observability spec https://github.com/grafana/tempo-operator/blob/cbe369b4983893cf50ce60201a1ed055f71bc344/apis/tempo/v1alpha1/tempostack_types.go#L141

Reply:
That would be inconsistent with the
```
<feature>:
  enabled: true
```
style of the CR. I'd prefer to migrate the `TempoStack`, maybe in the next CRD version? Shouldn't be too difficult to create a conversion webhook for this.

```
spec:                                    # TempoMonolithicSpec defines the desired state of TempoMonolithic.
  observability:                         # Observability defines observability configuration for the Tempo deployment
    metrics:                             # Metrics defines the metrics configuration of the Tempo deployment
      prometheusRules:                   # ServiceMonitors defines the PrometheusRule configuration
        enabled: false                   # Enabled defines if the operator should create PrometheusRules for this Tempo deployment
      serviceMonitors:                   # ServiceMonitors defines the ServiceMonitor configuration
        enabled: false                   # Enabled defines if the operator should create ServiceMonitors for this Tempo deployment
```

Reply:
so we agreed on using the enabled field which is better supported in tools like `kustomize` and `kubectl edit` (the empty structs are removed). Are we going to reuse some parts from the monolithic APIs?

Reply:
Yes, we can reuse a TLS struct, the multitenancy structs, ManagementStateType, LimitSpec and the storage secret.